net/quic/output.c

```diff
@@ -380,12 +381,12 @@ void quic_outq_retransmit(struct sock *sk)
 	}
 
 next:
-	skb = outq->retransmit_skb ?: skb_peek(head);
+	skb = outq->retransmit_skb ?: __skb_dequeue(head);
 	if (!skb)
 		return quic_outq_flush(sk);
-	__skb_unlink(skb, head);
```

Comment:
This change looks good but actually has problems:
`retransmit_skb` is also need to unlink from `retransmit_list`, as it will be linked back again after retransmit.

Revert it.